client/components/chart/index.js

```diff
@@ -81,7 +82,13 @@ class Chart extends Component {
 	componentDidUpdate( prevProps ) {
 		const { data } = this.props;
 		if ( ! isEqual( [ ...data ].sort(), [ ...prevProps.data ].sort() ) ) {
-			const orderedKeys = getOrderedKeys( this.props, this.state.orderedKeys );
+			/**
+			 * Only update the orderedKeys when data is present so that
+			 * selection may persist while requesting new data.
+			 */
+			const orderedKeys = data.length
```

Comment:
Might be better to move the `data.length` test into the `if` statement above?

Reply:
I tried that, but I still want `visibleData` to be updated when `data = []` so the displayed data series are removed. If `isRequesting` were truly accurate (see NOTE in original comment) this wouldn't be an issue because the placeholder would be shown instead. 

I'm thinking I'll leave it like this for now